lib/utils/fs_unix.go

```diff
@@ -19,7 +19,20 @@ limitations under the License.
 
 package utils
 
+import (
+	"os"
+	"syscall"
+)
+
 // On non-windows we just lock the target file itself.
 func getPlatformLockFilePath(path string) string {
 	return path
 }
+
+func getHardLinkCount(fi os.FileInfo) (bool, uint64) {
+	if statT, ok := fi.Sys().(*syscall.Stat_t); ok {
+		return true, statT.Nlink
+	} else {
+		return false, 0
+	}
```

Comment:
I'd write this slightly differently in order to eliminate the else and align the happy path to the left.

```
statT, ok := fi.Sys().(*syscall.Stat_t)
if !ok {
    return false, 0
}
return true, statT.Nlink
```

Reply:
This seems basically the same to me.  I tend to prefer avoiding `!` conditionals when logic for both states needs to be handled anyways.